app/service/search_index.rb

```diff
@@ -0,0 +1,17 @@
+class SearchIndex
+  def self.add(record)
+    if search_index_callbacks_enabled?
```

Comment:
Should there be some kind of error or alert if this isn't enabled? Error seems excessive, since we don't want a save to break because of indexing being disabled... It's OK if the answer is "nopes, we fail silently and get on with our lives." What do you think?